Gems/Atom/RPI/Code/Source/RPI.Reflect/Shader/ShaderAssetCreator.cpp

```diff
@@ -409,6 +414,21 @@ namespace AZ
             // copy the perAPIShaderData
             for (auto& perAPIShaderData : sourceShaderAsset.m_perAPIShaderData)
             {
+                // find the API in the list of supported APIs on this platform
+                AZStd::vector<RHI::ShaderPlatformInterface*>::const_iterator itFoundAPI = AZStd::find_if(
+                    platformInterfaces.begin(),
+                    platformInterfaces.end(),
+                    [&perAPIShaderData](const RHI::ShaderPlatformInterface* shaderPlatformInterface)
+                    {
+                        return perAPIShaderData.m_APIType == shaderPlatformInterface->GetAPIType();
+                    });
+
+                if (itFoundAPI == platformInterfaces.end())
+                {
+                    // the API is not supported on this platform, skip this entry
+                    continue;
+                }
+
```

Comment:
nit: I prefer this approach to the way it's done in `PrecompiledShaderBuilder::CreateJobs`

Reply:
I agree, updated.